plane2/src/drone/backend_manager.rs

```diff
@@ -222,6 +251,31 @@ impl BackendManager {
         }
     }
 
+    #[tracing::instrument]
+    fn create_metrics_handle(self: &Arc<Self>) {
+        let Ok(mut handle) = self.metrics_handle.lock() else {
+            tracing::error!("Metrics handle lock is poisoned");
+            return;
+        };
+        //cancel existing metrics gathering
+        handle.take();
+
+        let docker = self.docker.clone();
+        let id = self.container_id.clone();
+        let self_clone = self.clone();
+
+        *handle = Some(GuardHandle::new(async move {
+            while {
+                tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
+                true
```

Comment:
using [interval](https://docs.rs/tokio/latest/tokio/time/fn.interval.html) should give slightly more uniform time-steps